### PR TITLE
feat(notifications): support any node for subtitle & caption

### PR DESCRIPTION
**Pull request by jamesvclements**
Closes carbon-design-system/carbon-components-react#256

#### Changelog

**New**

- Allows an element as the subtitle and or caption for notifications

### CODE REVIEW

File: components/Notification.js [hunk 1]
```diff
@@ -101,9 +101,9 @@ class ToastNotification extends Component {
     className: PropTypes.string,
     kind: PropTypes.oneOf(['error', 'info', 'success', 'warning']).isRequired,
     title: PropTypes.string.isRequired,
-    subtitle: PropTypes.string.isRequired,
+    subtitle: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
```
Comment (joshblack) [inline components/Notification.js]: Would it be helpful instead for these props to just be `PropTypes.node`?

**Comment by joshblack** [comment]
Hi @jamesvclements! 👋 

Just wanted to say thanks so much for taking the time to contribute to `carbon-components-react` 🙏  we really appreciate any and all contributions.

Since it's past our typical working hours right now, will try and get this in front of some eyes tomorrow for reviews from some of the core members, in addition to anyone else who has the time to take a look 😄

File: components/Notification.js (hunk 1 above)
Comment (jamesvclements) [reply components/Notification.js]: @joshblack Yep that makes more sense. I'll swap them

**Comment by tw15egan** [comment]
Thanks @jamesvclements for tackling this! 👍 ✅

**Pull request merged**